### ISSUE TITLE
Invalid data for AMD 6700XT Card

### ISSUE THREAD

**Issue by marcostrama** [sp-hash/TeamBlackMiner#229]
TBM is showing invalid data for my AMD 6700XT card:

```
00:00:08 [2022-01-17 22:55:08.861]                                                                                 
00:00:08 [2022-01-17 22:55:08.861]                                                                                                                                                                                                                                                        
00:00:08 [2022-01-17 22:55:08.861] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT                                                                                                                                                                          
00:00:08 [2022-01-17 22:55:08.861] GPU0    gfx1031  OpenCL  0       71      0/0     0       0       0       0      
00:00:08 [2022-01-17 22:55:08.861] GPU1    3070     CUDA    1       284     53/0    84      1065    7900    109    
00:00:08 [2022-01-17 22:55:08.861] GPU2    3070     CUDA    1       284     53/0    92      1125    8050    118    
00:00:08 [2022-01-17 22:55:08.861]                                                                          227    
00:00:08 [2022-01-17 22:55:08.861]                                                                                                                                                                                                                                                        
00:00:08 [2022-01-17 22:55:08.861] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE                                                                                                                                                                           
00:00:08 [2022-01-17 22:55:08.861] GPU0    gfx1031  0.00 H/W      48.07 MH/s    47.89 MH/s    6/0/0        100.00% 
00:00:08 [2022-01-17 22:55:08.861] GPU1    3070     547.27 kH/W   61.84 MH/s    61.90 MH/s    5/0/0        100.00% 
00:00:08 [2022-01-17 22:55:08.861] GPU2    3070     518.10 kH/W   63.21 MH/s    63.41 MH/s    12/0/0       100.00% 
00:00:08 [2022-01-17 22:55:08.861]                  355.12 kH/W   173.12 MH/s   173.20 MH/s   23/0/0       100.00% 
00:00:08 [2022-01-17 22:55:08.861]                                                                                 
00:00:08 [2022-01-17 22:55:08.861] SHARES PER MINUTE: 2.88                                POOL HASHRATE: 191.66 MH/s        
```

GPU0 is showing the text "gfx1031", and wattage, temp, hash/w, etc is all zero.

GPU is a Powercolor Hellhound RX 6700XT

**Comment by marcostrama** [comment]
GPU data on hiveos:

```
GPU 0
0e:00.0
Radeon RX 6700 XT 12272 MB · PowerColor
Samsung GDDR6 · 113-D5122100_100
48.18 MH
51°
60°
66%
96 W
1250
650/700/1230
1075
```

**Comment by sp-hash** [comment]
Which version is this?

**Comment by marcostrama** [comment]
I tried with the last couple of versions, now it’s the 1.40

**Comment by chrishilbert** [comment]
I also have it for the XFX RX 6600s.  Reported a few days back but was closed out.

<img width="832" alt="image" src="https://user-images.githubusercontent.com/2332975/150023356-fac6cfe8-8019-42e1-82eb-afbf11b80ab1.png">

**Comment by chrishilbert** [comment]
The wattage inaccurate and temps are not registering correctly either for all my AMD cards from what I can tell.

**Comment by marcostrama** [comment]
Tested with the new 1.42 and still with invalid data

**Comment by chrishilbert** [comment]
Look at fan speeds on my cards. They're all set at 60%.

**Comment by sp-hash** [comment]
Can you ssh into the rig and run the following command please

rocm-smi --json --showbus --showtemp --showpower --showclocks --showfan

**Comment by sp-hash** [comment]
Also try to upgrade the packages with

sudo apt install update

**Comment by chrishilbert** [comment]
Device order is fine now and matches HiveOS.  Apt packages are all up-to-date, run apt update a lot.

rocm-smi output:

{"card3": {"fclk clock speed": "1555Mhz", "sclk clock speed": "900Mhz", "dcefclk clock level": "1", "Fan Speed (%)": "60", "socclk clock level": "1", "fclk clock level": "1", "PCI Bus": "0000:0a:00.0", "mclk clock level": "3", "Temperature (Sensor mem) (C)": "54.0", "Fan Speed (level)": "153", "mclk clock speed": "950Mhz", "socclk clock speed": "1200Mhz", "pcie clock level": "1", "sclk clock level": "1", "Average Graphics Package Power (W)": "44.0", "Temperature (Sensor junction) (C)": "42.0", "dcefclk clock speed": "738Mhz", "Temperature (Sensor edge) (C)": "37.0", "pcie clock speed": "5.0GT/s, x1 619Mhz"}, "card2": {"fclk clock speed": "1805Mhz", "sclk clock speed": "1200Mhz", "dcefclk clock level": "1", "Fan Speed (%)": "60", "socclk clock level": "1", "fclk clock level": "1", "PCI Bus": "0000:07:00.0", "mclk clock level": "3", "Temperature (Sensor mem) (C)": "50.0", "Fan Speed (level)": "153", "mclk clock speed": "1074Mhz", "socclk clock speed": "1200Mhz", "pcie clock level": "1", "sclk clock level": "1", "Average Graphics Package Power (W)": "91.0", "Temperature (Sensor junction) (C)": "45.0", "dcefclk clock speed": "872Mhz", "Temperature (Sensor edge) (C)": "41.0", "pcie clock speed": "5.0GT/s, x1 619Mhz"}, "card1": {"fclk clock speed": "1551Mhz", "sclk clock speed": "1325Mhz", "dcefclk clock level": "0", "Fan Speed (%)": "60", "socclk clock level": "0", "fclk clock level": "0", "PCI Bus": "0000:03:00.0", "mclk clock level": "3", "Temperature (Sensor mem) (C)": "80.0", "Fan Speed (level)": "153", "mclk clock speed": "1059Mhz", "socclk clock speed": "872Mhz", "pcie clock level": "1", "sclk clock level": "1", "Average Graphics Package Power (W)": "111.0", "Temperature (Sensor junction) (C)": "49.0", "dcefclk clock speed": "417Mhz", "Temperature (Sensor edge) (C)": "43.0", "pcie clock speed": "5.0GT/s, x1 619Mhz"}, "card6": {"fclk clock speed": "1556Mhz", "sclk clock speed": "900Mhz", "dcefclk clock level": "1", "Fan Speed (%)": "60", "socclk clock level": "1", "fclk clock level": "1", "PCI Bus": "0000:13:00.0", "mclk clock level": "3", "Temperature (Sensor mem) (C)": "52.0", "Fan Speed (level)": "153", "mclk clock speed": "950Mhz", "socclk clock speed": "1200Mhz", "pcie clock level": "1", "sclk clock level": "1", "Average Graphics Package Power (W)": "41.0", "Temperature (Sensor junction) (C)": "40.0", "dcefclk clock speed": "738Mhz", "Temperature (Sensor edge) (C)": "35.0", "pcie clock speed": "5.0GT/s, x1 619Mhz"}, "card5": {"fclk clock speed": "1554Mhz", "sclk clock speed": "935Mhz", "dcefclk clock level": "1", "Fan Speed (%)": "60", "socclk clock level": "1", "fclk clock level": "1", "PCI Bus": "0000:10:00.0", "mclk clock level": "3", "Temperature (Sensor mem) (C)": "50.0", "Fan Speed (level)": "153", "mclk clock speed": "1079Mhz", "socclk clock speed": "1200Mhz", "pcie clock level": "1", "sclk clock level": "1", "Average Graphics Package Power (W)": "52.0", "Temperature (Sensor junction) (C)": "42.0", "dcefclk clock speed": "960Mhz", "Temperature (Sensor edge) (C)": "37.0", "pcie clock speed": "5.0GT/s, x1 619Mhz"}, "card4": {"fclk clock speed": "1556Mhz", "sclk clock speed": "900Mhz", "dcefclk clock level": "1", "Fan Speed (%)": "60", "socclk clock level": "1", "fclk clock level": "1", "PCI Bus": "0000:0d:00.0", "mclk clock level": "3", "Temperature (Sensor mem) (C)": "52.0", "Fan Speed (level)": "153", "mclk clock speed": "950Mhz", "socclk clock speed": "1200Mhz", "pcie clock level": "1", "sclk clock level": "1", "Average Graphics Package Power (W)": "43.0", "Temperature (Sensor junction) (C)": "43.0", "dcefclk clock speed": "738Mhz", "Temperature (Sensor edge) (C)": "39.0", "pcie clock speed": "5.0GT/s, x1 619Mhz"}}

**Comment by chrishilbert** [comment]
Here's what HiveOS shows:

<img width="1219" alt="image" src="https://user-images.githubusercontent.com/2332975/150140764-c068348b-c5de-4760-8336-9aa1064113e2.png">

**Comment by sp-hash** [comment]
thx.

Fixed in v1.43

**Comment by marcostrama** [comment]
Nice, thanks!

**Comment by marcostrama** [comment]
@sp-hash installed the new version but still with invalid data for me. Output of rocm-smi:

```
root@Rig1:~# rocm-smi --json --showbus --showtemp --showpower --showclocks --showfan                                                                                                                                                                                                      
{"card6": {"fclk clock speed": "1551Mhz", "sclk clock speed": "1250Mhz", "dcefclk clock level": "1", "Fan Speed (%)": "23", "socclk clock level": "1", "fclk clock level": "0", "PCI Bus": "0000:0e:00.0", "mclk clock level": "3", "Temperature (Sensor mem) (C)": "50.0", "Fan Speed (le
vel)": "61", "mclk clock speed": "1074Mhz", "socclk clock speed": "1200Mhz", "pcie clock level": "1", "sclk clock level": "1", "Average Graphics Package Power (W)": "29.0", "Temperature (Sensor junction) (C)": "48.0", "dcefclk clock speed": "960Mhz", "Temperature (Sensor edge) (C)"
: "46.0", "pcie clock speed": "5.0GT/s, x1 619Mhz"}} 
```

**Comment by sp-hash** [comment]
fixed in v1.45
